### ISSUE TITLE
Gene search hangs program when sequential searches are not separated by a reset button press

### ISSUE THREAD

**Issue by astrasb** [astrasb/Strongyloides_Bioinformatics#40]
In the gene-wise browsing tab, the code hangs if you try and submit a new search without using the clear button first. This is particularly pernicious in the deployed version on ShinyApps.io, as it seems to stall the entire program such that all other responsive elements become non-functional. 

To make the program more robust for users, this issue should probably be looked at more closely.

**Comment by astrasb** [comment]
I maybe solved one part of this by adjusting the shinyapps.io settings to give the program more processing time before the io shuts the process down. The main part of this issue is just that if the gene expression plot for the second search is also going to show the All Genes heatmap (and thus the dropdown menu selection doesn't change), the plot doesn't get produced b/c the input from the dropdown menu doesnt seem to change.

Maybe this can be solved by adding a dependency somewhere?

**Comment by astrasb** [comment]
Today I confirmed that this is a particular issue when doing multiple searches with gene names, and I'm not confident that the clear button is working as expected.

**Comment by astrasb** [comment]
Note: underlying data appears to be there, it's just that the gene expression plots are not being appropriately generated. But the genelist is being regenerated, and correctly populated.

**Comment by astrasb** [comment]
After some detective work, it turns out culprit was in part the shinycssloaders::withSpinner code. Short version: certain outputs including plotly don't send the appropriate javascript signal to the ui when their values change. See a discussion of a similar problem in this shinycssloaders github issue:  https://github.com/daattali/shinycssloaders/issues/17#issuecomment-623044587

To fix this, I have removed the shinycssloaders code from the plots of gene expression. 

I also fixed one of the stalling events by switching from an observeEvent triggered by a change in the gene pulldown menu, to a normal observer that will automatically update whenever either the pulldown or the genelist are changed.
